Add support for custom BusyBox image

tomheijmans opened the pull request:
This pull request adds support for using a custom BusyBox image as requested in issue #127. I think it integrates with the already existing customization's for the dashboard and node image.

Example usage using the `general.defaultRepo` option:
```yaml
  general:
    defaultRepo: "mycustomrepo.cr"
```
Or a custom image can be specified using the following example:
```yaml
 busyBox:
    # specify version
    version: 1.27.2-buildcustom
    # or specify a totally different image
    image: "mycustomrepo.cr/mycustombusybox:myversion"
```

swoehrl-mw reviewed (changes requested):
Hi @tomheijmans. Thank you for your contribution.
I have several points:
* I don't like that the option is named `busyBox`. Yes, we are using a busybox image, but looking at it abstractly we have a helper container. So I'd prefer to name this `initHelper`. @idanl21 @prudhvigodithi What do you think of the name?
* Can you please add a unittest for the new functionality (probably best in `pkg/builders/cluster_test.go`)?
* Please document the new fields in `docs/designs/crd.md` and add a short section to the userguide (`docs/userguide/main.md`)

tomheijmans commented:
@swoehrl-mw thank you for your feedback on this. I Agree the name of option (`busyBox`) feels to technical. I renamed it to `initHelper` as you suggested. Added multiple test cases for this change which I hope make sense as I'm not a GO expert.

Regards Tom

prudhvigodithi commented on docs/userguide/main.md:
````diff
@@ -311,6 +311,27 @@ Or custom ones, for example that Aiven plugin for prometheus-exporter:
 ```
 
+## Custom image repository
+A custom image repository can be configured as needed (for example an air gapped environment). Globally used image repository is set using the `DefaultRepo` option:
````
NIT: please change the heading to Custom init helper and a line that says this image will be used during the cluster setup. :)

tomheijmans replied:
I made some adjustments in it to make it more clear. Does it make sense to you?

prudhvigodithi commented:
Thanks @tomheijmans  LGTM. @swoehrl-mw and @idanl21 let me know if you are ok to merge?
Thank you

idanl21 commented:
> Hi @tomheijmans. Thank you for your contribution. I have several points:
> 
> * I don't like that the option is named `busyBox`. Yes, we are using a busybox image, but looking at it abstractly we have a helper container. So I'd prefer to name this `initHelper`. @idanl21 @prudhvigodithi What do you think of the name?
> * Can you please add a unittest for the new functionality (probably best in `pkg/builders/cluster_test.go`)?
> * Please document the new fields in `docs/designs/crd.md` and add a short section to the userguide (`docs/userguide/main.md`)

Agree with the naming, didn't see the section under user-guide yet.

tomheijmans commented on docs/userguide/main.md:
````diff
@@ -311,6 +311,27 @@ Or custom ones, for example that Aiven plugin for prometheus-exporter:
+      defaultRepo: "mycustomrepo.cr"
+```
+
+Alternatively, a custom image can be specified for each image separately. For example, the initHelper image is configured as follows:
````
I made some adjustments in it to make it more clear. Does it make sense to you?

idanl21 replied:
you can add at the beginning, ' by default OpensearchOperator creates init container with that image (public.ecr.aws/opsterio/busybox:1.27.2-buildx) according to change that behaviour and use a custom image, follow the instructions '
I think that way there will be no chance for confusion.

tomheijmans replied:
Thanks for your suggestion. Changed it.

swoehrl-mw reviewed (approved):
LGTM.
@tomheijmans Can you plase update your branch and resolve the conflicts? Then we can merge.

prudhvigodithi commented:
Hey @tomheijmans can you please resolve the conflicts?
Thank you

tomheijmans commented:
@swoehrl-mw, @prudhvigodithi updated this branch with changes from main. Should all be up to date now.  
Thanks for accepting my contribution :)
